queries.sql

```diff
@@ -188,4 +188,124 @@ FROM animals A
 JOIN owners O
 ON A.owner_id = O.id
 GROUP BY O.full_name
-ORDER BY animal DESC;
\ No newline at end of file
+ORDER BY animal DESC;
+
+/*MANY-TO-MANY QUERIES*/
+
+SELECT
+A.name AS animal_name,
+vets.name AS vet_name,
+visits.date_of_visit
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC;
+
+SELECT DISTINCT
+vets.name,
+COUNT(A.name) AS animals_attended
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+GROUP BY vets.name;
+
+SELECT
+vets.name AS vet_name,
+species.name AS specialization
+FROM specializations S
+JOIN species
+ON species.id = S.species_id
+RIGHT JOIN vets
+ON vets.id = S.vet_id;
+
+SELECT
+A.name AS animal_name,
+vets.name AS vet_name,
+visits.date_of_visit
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE visits.date_of_visit >= '2020-04-01'
+AND visits.date_of_visit <= '2020-08-30'
+AND vets.name = 'Stephanie Mendez';
+
+SELECT
+A.name,
+COUNT(A.name) AS visits
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+GROUP BY A.name
+ORDER BY visits DESC;
+
+SELECT
+A.name AS animal_name,
+vets.name AS vet_name,
+visits.date_of_visit
+FROM visits
+JOIN animals A
+ON A.id = visits.animal_id
+JOIN vets
+ON vets.id = visits.vet_id
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC;
```

Comment:
- [ ] Great job on implementing this query! 🎉 However, according to the guidelines, it should return just one row, indicating the first animal visited. 🐾🗺️ Let's make the necessary adjustments to ensure it functions correctly. 🛠️👍